### PR TITLE
source-microsoft-lists contribution from bishalbera

### LINKED ISSUE #46615: [connector-request] Microsoft Lists
This is a **suggestion** to create a connector for **Microsoft Lists** using the UI Connector Builder.
The website for the connector is https://www.microsoft.com/en-us/microsoft-365/microsoft-lists.

If you want to try, claim this issue and start working on it following the steps below.

### Steps:
1. Comment in the issue and wait to be assigned to start working on it.
2. Map the API endpoints (get approval to move to 3 step)
3. Describe steps to get credentials.
4. Create the connector using the UI Builder

> [!IMPORTANT]
> Airbyte doesn't have sandbox credentials for this connector. You must have access/credentials to the service provider to create the connector.

> [!CAUTION]
> This is a **suggestion** and there may be cases where creating the connector using the Builder won't be possible. To minimize the risk of investing a lot of time directly in creating the connector, we strongly recommend to follow steps above.

### PR DESCRIPTION
**Pull request by bishalbera**
## What

This PR adds a new connector for source Microsoft Lists (source-microsoft-lists).

Contributor-provided connector description:

Microsoft Lists connector enables seamless data integration and synchronization between Microsoft Lists and other destination. The connector leverages Microsoft Graph API to retrieve list items efficiently, ensuring smooth workflows and real-time data accessibility
## Details

<details><summary>Expand for list of spec properties and streams</summary>

### Configuration

| Input | Type | Description | Default Value |
|-------|------|-------------|---------------|
| `site_id` | `string` | Site Id.  |  |
| `client_id` | `string` | Client ID.  |  |
| `client_secret` | `string` | Client secret.  |  |
| `application_id_uri` | `string` | Application Id URI.  |  |
| `tenant_id` | `string` | Tenant Id.  |  |
| `domain` | `string` | Domain.  |  |

### Streams
| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
|-------------|-------------|------------|---------------------|----------------------|
| lists | id | DefaultPaginator | ✅ |  ❌  |
| listcontenttypes | id | DefaultPaginator | ✅ |  ❌  |
| listitems |  | DefaultPaginator | ✅ |  ❌  |
| items |  | DefaultPaginator | ✅ |  ❌  |
| columnvalues | id | DefaultPaginator | ✅ |  ❌  |

</details>

## Reviewer checklist
- [ ] Run `/format-fix` to resolve any formatting errors
- [ ] Help the contributor upload an icon that follows our icon design guidelines
- [ ] Click into the CI workflows that wait for a maintainer to run them, which should trigger CI runs

### CODE REVIEW

File: docs/integrations/sources/microsoft-lists.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+## Configuration
+
+| Input | Type | Description | Default Value |
+|-------|------|-------------|---------------|
+| `client_id_2` | `string` | Client ID.  |  |
+| `client_secret_2` | `string` | Client secret.  |  |
+| `application_id_uri` | `string` | Application Id URI.  |  |
+| `site_id` | `string` | Site Id.  |  |
+| `tenant_id` | `string` | Tenant Id.  |  |
+| `domain` | `string` | Domain.  |  |
```
Comment (marcosmarxm) [inline docs/integrations/sources/microsoft-lists.md:13]: Provide more information about the parameters how to get them and configure them in a proper section in docs.

File: docs/integrations/sources/microsoft-lists.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+
+## Streams
+| Stream Name | Primary Key | Pagination | Supports Full Sync | Supports Incremental |
+|-------------|-------------|------------|---------------------|----------------------|
+| lists | id | No pagination | ✅ |  ❌  |
+| listcontenttypes | id | No pagination | ✅ |  ❌  |
+| listitems |  | No pagination | ✅ |  ❌  |
+| items |  | No pagination | ✅ |  ❌  |
+| columnvalues | id | No pagination | ✅ |  ❌  |
```
Comment (marcosmarxm) [inline docs/integrations/sources/microsoft-lists.md:22]: Without pagination it won't be possible to accept the connector. Please read more how Microsoft do pagination: https://learn.microsoft.com/en-us/graph/paging?tabs=http

**Review by marcosmarxm — CHANGES_REQUESTED** [review]

File: docs/integrations/sources/microsoft-lists.md (hunk 1 above)
Comment (bishalbera) [reply docs/integrations/sources/microsoft-lists.md:13]: ok

File: docs/integrations/sources/microsoft-lists.md (hunk 2 above)
Comment (bishalbera) [reply docs/integrations/sources/microsoft-lists.md:22]: ya I also was going to add the pagination to this one(after entra id) tomorrow .

Comment (bishalbera) [reply docs/integrations/sources/microsoft-lists.md]: @marcosmarxm I've modified the doc and added the pagination. Please check now and tell me if I've done anything wrong

**Review by marcosmarxm — CHANGES_REQUESTED** [review]
Please check the comment in the other Microsoft contribution https://github.com/airbytehq/airbyte/pull/46730#pullrequestreview-2376280455

**Comment by marcosmarxm** [comment]
Hello 👋 Due to the numerous contributions happening, I decided to create this [view](https://github.com/orgs/airbytehq/projects/104/views/3) to help you know when I plan to review your contribution. If your connector has some limited-time credential feel free to reach me out.

**Review by marcosmarxm — APPROVED** [review]

**Pull request merged**